tests/test_modules/test_web/test_system_websocket.py

```diff
@@ -26,31 +26,136 @@ def tearDown(self):
         self.process.stop(timeout=1)
 
     @gen.coroutine
-    def send_message(self):
+    def send_message(self, req):
         conn = yield websocket_connect("ws://localhost:%s/ws" % self.socket)
-        req = dict(
-            typeid="malcolm:core/Post:1.0",
-            id=0,
-            path=["hello", "greet"],
-            parameters=dict(
-                name="me"
-            )
-        )
         conn.write_message(json.dumps(req))
         resp = yield conn.read_message()
         resp = json.loads(resp)
         self.result.put(resp)
         conn.close()
 
+    @gen.coroutine
+    def send_non_JSON_message(self):
```

Comment:
Can you merge this with the previous function please. I suggest

`def send_message(self, req, json_convert=True):`